src/appHost.tsx

```diff
@@ -511,6 +526,14 @@ miss: ${memoizedWithMissHit.miss}
         throw new Error(`API '${slotKeyToName(key)}' doesn't exist.`)
     }
 
+    function executeWhenFree(key: string, callback: () => void): void {
+        if (isInstallingEntryPoints) {
+            entryPointsInstallationEndCallbacks.set(key, callback)
```

Comment:
Why do we need these callbacks saved?
Won't the state update on install end be enough?

Reply:
1. AFAIK we do not want to relay on store update to cause re render when contributing new item.
2. Can we guarantee that `setInstalledShellNames` is called and updates the store in all possible installation flows?

Reply:
After discussing this in person it was decided that it is better not to relay on the impl details of the install shells flow (meaning relay on a dispatch action that we know happens at the end of the flow), especially that we know that local state update + store update will not actually cause double re-render of all contributed item (so the cost of the additional local state update is low). 
So leaving it as is.